pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
```

Comment:
We're going to need to support pagination (#2349) and incremental updates, for both writes and reads, for large sets. A solution would be to make /endpoints a list of lists, which naively would make initial creation require a lot more requests, but which could be mitigated by supporting Batches of requests or POSTs of lists.